src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/AssemblyIdentityMustMatch.cs

```diff
@@ -0,0 +1,106 @@
+
+            if (leftAssemblyName != rightAssemblyName)
+            {
+                differences.Add(CreateDifference(Resources.AssemblyNameDoesNotMatch, leftAssemblyName, rightAssemblyName, leftName, rightName, rightIdentity));
+            }
+
+            if (leftAssemblyCulture != rightAssemblyCulture)
+            {
+                differences.Add(CreateDifference(Resources.AssembyCultureDoesNotMatch, leftAssemblyCulture, rightAssemblyCulture, leftName, rightName, rightIdentity));
+            }
+ 
+            if (rightAssemblyVersion < leftAssemblyVersion)
+            {
+                differences.Add(CreateDifference(Resources.AssembyVersionDoesNotMatch, rightAssemblyVersion.ToString(), leftAssemblyVersion.ToString(), rightName, leftName, rightIdentity));
+            }
+            else if (Settings.StrictMode && leftAssemblyVersion < rightAssemblyVersion)
+            {
+                differences.Add(CreateDifference(Resources.AssembyVersionDoesNotMatch, leftAssemblyVersion.ToString(), rightAssemblyVersion.ToString(), leftName, rightName, rightIdentity));
+            }
+
+            bool isLeftRetargetable = (left.Identity.Flags & AssemblyNameFlags.Retargetable) != 0;
+            bool isRightRetargetable = (right.Identity.Flags & AssemblyNameFlags.Retargetable) != 0;
+            if (!leftAssemblyPublicKeyToken.IsEmpty && !isLeftRetargetable && !leftAssemblyPublicKeyToken.SequenceEqual(rightAssemblyPublicKeyToken))
+            {
+                differences.Add(CreateDifference(
+                    Resources.AssemblyPublicKeyTokenDoesNotMatch,
+                    GetStringRepresentation(leftAssemblyPublicKeyToken), 
+                    GetStringRepresentation(rightAssemblyPublicKeyToken), 
+                    leftName, 
+                    rightName,
+                    rightIdentity));
+            }
+            else if (Settings.StrictMode && !rightAssemblyPublicKeyToken.IsEmpty && !isRightRetargetable)
+            {
+                differences.Add(CreateDifference(
+                    Resources.AssemblyPublicKeyTokenDoesNotMatch,
+                    GetStringRepresentation(rightAssemblyPublicKeyToken), 
+                    GetStringRepresentation(leftAssemblyPublicKeyToken), 
+                    rightName,
+                    leftName,
+                    leftIdentity));
+            }
+        }
+
+        private static string GetStringRepresentation(ReadOnlySpan<byte> publicKeyToken)
+        {
+            if (publicKeyToken.IsEmpty)
+                return "null";
+
+            StringBuilder sb = new StringBuilder();
+            foreach (var b in publicKeyToken)
+            {
+                sb.Append(b.ToString("x2"));
+            }
+            return sb.ToString();
+        }
+
+        private CompatDifference CreateDifference(string format, string leftProperty, string rightProperty, string leftName, string rightName, AssemblyIdentity identity) =>
```

Comment:
Maybe rename this to, `CreateIdentityDifference`? as this is not used by the name difference?